src/simulate.jl

```diff
@@ -0,0 +1,89 @@
+using DifferentialEquations
+const DE = DifferentialEquations
+
+function simulate(cp::AbstractNeuralNetworkControlProblem, args...; kwargs...)
+    ivp = plant(cp)
+    network = controller(cp)
+    st_vars = state_vars(cp)
+    n = length(st_vars)
+    X₀ = LazySets.Projection(initial_state(ivp), st_vars)
+    ctrl_vars = control_vars(cp)
+    δ = period(cp)
+    time_span = _get_tspan(args...; kwargs...)
+    trajectories = get(kwargs, :trajectories, 10)
+    inplace = get(kwargs, :inplace, true)
+
+    t = tstart(time_span)
+    iterations = ceil(Int, diam(time_span) / δ)
+
+    # preallocate
+    extended = Vector{Vector{Float64}}(undef, trajectories)
+    simulations = Vector{EnsembleSolution}(undef, iterations)
+    all_controls = Vector{Vector{Vector{Float64}}}(undef, iterations)
```

Comment:
it's not easy to see what do these nested Vector correspond to, we can add some auxiliary structs (later, not critical)